src/levanter/main/train_lm.py

```diff
@@ -16,7 +16,7 @@
 from levanter import callbacks
 from levanter.checkpoint import load_checkpoint
 from levanter.compat.hf_checkpoints import HFCompatConfig, save_hf_checkpoint_callback
-from levanter.data.text import CausalLmDataset, LMDatasetConfig, LMMixtureDatasetConfig
+from levanter.data.text import CausalLmDataset, LMDatasetConfig, LMMixtureDatasetConfig, LMSupervisedDatasetConfig
```

Comment:
Should you maybe make a different file? I think train_lm is typically expected within levanter to be reserved for pre-training and is specific to the autoregressive loss. I think there's separate files for fine-tuning alpaca for example. 

I'm not sure what the convention is cc: @dlwh

Reply:
For supervised finetuning, we maybe should make a separate file but the changes in this file are for measuring eval loss on a supervised task.

Reply:
Got it LGTM